### PR TITLE
feat: integrated backend discussions restriction with MFE

### PR DESCRIPTION
**Pull request by ayesha-waris**
[INF-828](https://2u-internal.atlassian.net/browse/INF-828)
### Description

Integrated already build UI with the API change in [INF-826](https://2u-internal.atlassian.net/browse/INF-826)

#### Post-merge Checklist

* [ ] Deploy the changes to prod after verifying on stage or ask **@openedx/edx-infinity** to do it. 
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

File: src/discussions/data/hooks.js [hunk 1]
```diff
@@ -192,17 +190,16 @@ export const useCurrentDiscussionTopic = () => {
...
   const isUserAdmin = useSelector(selectUserIsStaff);
   const userHasModerationPrivileges = useSelector(selectUserHasModerationPrivileges);
   const isUserGroupTA = useSelector(selectUserIsGroupTa);
   const isCourseAdmin = useSelector(selectIsCourseAdmin);
   const isCourseStaff = useSelector(selectIsCourseStaff);
   const isPrivileged = isUserAdmin || userHasModerationPrivileges || isUserGroupTA || isCourseAdmin || isCourseStaff;
-  const isInBlackoutDateRange = useMemo(() => inBlackoutDateRange(blackoutDateRange), [blackoutDateRange]);
 
-  return (!(isInBlackoutDateRange) || (isPrivileged));
+  return (isPostingEnabled || (isPrivileged));
```
Comment (awais-ansari) [inline src/discussions/data/hooks.js:202]: (isPostingEnabled || isPrivileged)

File: src/discussions/data/hooks.js [hunk 2]
```diff
@@ -192,17 +190,16 @@ export const useCurrentDiscussionTopic = () => {
   return null;
 };
 
-export const useUserCanAddThreadInBlackoutDate = () => {
-  const blackoutDateRange = useSelector(selectBlackoutDate);
+export const useUserCanAddThreadInPostingRestriction = () => {
```
Comment (awais-ansari) [inline src/discussions/data/hooks.js:193]: useUserCanAddThreadInPostingRestriction => useUserPostingEnabled

File: src/discussions/discussions-home/BlackoutInformationBanner.jsx [hunk 3]
```diff
@@ -1,23 +1,18 @@
...
 
 import { useIntl } from '@edx/frontend-platform/i18n';
 import { PageBanner } from '@edx/paragon';
 
-import { selectBlackoutDate } from '../data/selectors';
+import { selectIsPostingEnabled } from '../data/selectors';
 import messages from '../messages';
-import { inBlackoutDateRange } from '../utils';
 
 const BlackoutInformationBanner = () => {
```
Comment (awais-ansari) [inline src/discussions/discussions-home/BlackoutInformationBanner.jsx:11]: change the name of function or anywhere we area using blockoutDates.

File: src/discussions/discussions-home/BlackoutInformationBanner.jsx [hunk 4]
```diff
@@ -1,23 +1,18 @@
...
   const intl = useIntl();
-  const blackoutDate = useSelector(selectBlackoutDate);
+  const isDiscussionsBlackout = !useSelector(selectIsPostingEnabled);
```
Comment (awais-ansari) [inline src/discussions/discussions-home/BlackoutInformationBanner.jsx:13]: IsPostingEnabled =  useSelector(selectIsPostingEnabled);

File: src/discussions/data/slices.js [hunk 5]
```diff
@@ -18,6 +17,7 @@ const configSlice = createSlice({
     isCourseStaff: false,
     isUserAdmin: false,
     learnersTabEnabled: false,
+    isPostingEnabled: true,
```
Comment (awais-ansari) [inline src/discussions/data/slices.js:20]: make it false.

File: src/discussions/discussions-home/BlackoutInformationBanner.test.jsx [hunk 6]
```diff
@@ -56,13 +56,11 @@ describe('Blackout Information Banner', () => {
...
 
   test.each([
-    { blackouts: [], visibility: false },
-    { blackouts: ['2021-12-31T10:15', '2021-12-31T10:20'], visibility: false },
-    { blackouts: [{ start: activeStartDate, end: activeEndDate }], visibility: true },
-    { blackouts: [{ start: activeEndDate, end: activeEndDate }], visibility: false },
-  ])('Test Blackout Banner is visible on app load if blackout date is active', async ({ blackouts, visibility }) => {
+    { isPostingEnabled: false, visibility: true },
+    { isPostingEnabled: true, visibility: false },
+  ])('Test Blackout Banner is visible on app load if posting is disabled', async ({ isPostingEnabled, visibility }) => {
```
Comment (awais-ansari) [inline src/discussions/discussions-home/BlackoutInformationBanner.test.jsx:61]: update the description

File: src/discussions/common/ActionsDropdown.jsx [hunk 7]
```diff
@@ -41,12 +41,12 @@ const ActionsDropdown = ({
     }
   }, [actionHandlers]);
 
-  // Find and remove edit action if in blackout date range.
+  // Find and remove edit action if in Posting restriction date range or postimg is disabled.
```
Comment (awais-ansari) [inline src/discussions/common/ActionsDropdown.jsx:44]: update the description

**Review by awais-ansari — CHANGES_REQUESTED** [review]

**Review by awais-ansari — APPROVED** [review]

**Review by sundasnoreen12 — APPROVED** [review]

**Pull request merged**